### PR TITLE
Custom span name

### PR DESCRIPTION
**Pull request by avnandu**
Can customize span name as per user choice

### CODE REVIEW

File: opentracing-redis-common/pom.xml [hunk 1]
```diff
@@ -14,15 +14,24 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>opentracing-redis-parent</artifactId>
+        <groupId>io.opentracing.contrib</groupId>
+        <version>0.0.4-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>opentracing-redis-common</artifactId>
```
Comment (desumera) [inline opentracing-redis-common/pom.xml]: FYI, they might give feedback that the formatting changed here without any "real" changes - not sure.

File: opentracing-redis-common/src/main/java/io/opentracing/contrib/redis/common/RedisSpanNameProvider.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+public class RedisSpanNameProvider {
+
+    public static Function<String, String> OPERATION_NAME = (operationName) -> ((operationName == null) ? "unknown" : operationName);
+
+
+    public static Function<String, String> PREFIX_OPERATION_NAME(final String prefix) {
+        return (operationName) ->
+            ((prefix == null) ? "" : prefix)
+                    + ((operationName == null) ? "unknown" : operationName);
```
Comment (desumera) [inline opentracing-redis-common/src/main/java/io/opentracing/contrib/redis/common/RedisSpanNameProvider.java]: Can we add tests for these? I think I'd be nitpicking to ask for these expressions to be in their own method (for sake of nice naming); having tests would illustrate the same point. And then you'd also have tests 😝

File: opentracing-redis-common/src/main/java/io/opentracing/contrib/redis/common/TracingHelper.java [hunk 3]
```diff
@@ -18,150 +18,155 @@
...
 import io.opentracing.noop.NoopSpan;
 import io.opentracing.tag.Tags;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+
+import java.util.*;
```
Comment (desumera) [inline opentracing-redis-common/src/main/java/io/opentracing/contrib/redis/common/TracingHelper.java]: No `*` imports!

File: opentracing-redis-common/src/main/java/io/opentracing/contrib/redis/common/RedisSpanNameProvider.java (hunk 2 above)
Comment (desumera) [reply opentracing-redis-common/src/main/java/io/opentracing/contrib/redis/common/RedisSpanNameProvider.java]: Actually, it looks like you're testing the Provider in your TracingHelperTest. Let's keep those separate. That way, we can extensively test the Provider, and what's being tested is also clear.

File: opentracing-redis-common/src/main/java/io/opentracing/contrib/redis/common/TracingHelper.java [hunk 4]
```diff
@@ -18,150 +18,155 @@
...
+    private final boolean traceWithActiveSpanOnly;
+    private Function<String, String> spanNameProvider;
 
-  private SpanBuilder builder(String operationName) {
-    return tracer.buildSpan(operationName)
-        .withTag(Tags.COMPONENT.getKey(), COMPONENT_NAME)
-        .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CLIENT)
-        .withTag(Tags.DB_TYPE.getKey(), "redis");
-  }
+    public TracingHelper(Tracer tracer, boolean traceWithActiveSpanOnly) {
```
Comment (desumera) [inline opentracing-redis-common/src/main/java/io/opentracing/contrib/redis/common/TracingHelper.java]: Figure out how all this got re-ordered

**Review by desumera — CHANGES_REQUESTED** [review]

**Comment by eleduardo** [comment]
Sorry to do this, we might need to go to the master branch and start again to avoid indentation and re-ordering issues :(. Not sure they will take this PR if we keep these...

File: .settings.xml [hunk 5]
```diff
@@ -15,29 +15,29 @@
...
+        <server>
+            <id>jfrog-snapshots</id>
+            <username>${env.BINTRAY_USER}</username>
+            <password>${env.BINTRAY_KEY}</password>
+        </server>
+        <server>
+            <id>github.com</id>
+            <username>${env.GH_USER}</username>
+            <password>${env.GH_TOKEN}</password>
+        </server>
```
Comment (desumera) [inline .settings.xml]: extra spaces!!!!

File: opentracing-redis-common/src/main/java/io/opentracing/contrib/redis/common/TracingHelper.java [hunk 6]
```diff
@@ -13,32 +13,43 @@
...
+import io.opentracing.noop.NoopSpan;
+import io.opentracing.tag.Tags;
 
 public class TracingHelper {
 
+  private static final String COMPONENT_NAME = "java-redis";
   private final Tracer tracer;
   private final boolean traceWithActiveSpanOnly;
-  private static final String COMPONENT_NAME = "java-redis";
+  private Function<String, String> spanNameProvider;
```
Comment (desumera) [inline opentracing-redis-common/src/main/java/io/opentracing/contrib/redis/common/TracingHelper.java]: Should this be final?

File: opentracing-redis-common/src/test/java/io/opentracing/contrib/redis/common/RedisSpanNameProviderTest.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  Function<String, String> prefixSpanName;
+
+  @Before
+  public void setUp() {
+    prefixSpanName = RedisSpanNameProvider.PREFIX_OPERATION_NAME(prefix);
+  }
+
+  @Test
+  public void testGetPrefix() {
+    //span = (MockSpan) (helper.buildSpan(get));
```
Comment (desumera) [inline opentracing-redis-common/src/test/java/io/opentracing/contrib/redis/common/RedisSpanNameProviderTest.java]: Pls remove commented line

File: opentracing-redis-lettuce/src/main/java/io/opentracing/contrib/redis/lettuce/TracingRedisAsyncCommands.java [hunk 8]
```diff
@@ -55,21 +61,19 @@
...
+import static io.opentracing.contrib.redis.common.TracingHelper.nullable;
+import static io.opentracing.contrib.redis.common.TracingHelper.onError;
 
 public class TracingRedisAsyncCommands<K, V> implements RedisAsyncCommands<K, V> {
 
   private final RedisAsyncCommands<K, V> commands;
   private final Tracer tracer;
   private final boolean traceWithActiveSpanOnly;
   private final TracingHelper helper;
+  private Function<String, String> redisSpanNameProvider;
```
Comment (desumera) [inline opentracing-redis-lettuce/src/main/java/io/opentracing/contrib/redis/lettuce/TracingRedisAsyncCommands.java:75]: Should this be final?

**Review by desumera — APPROVED** [review]
Looks good! When you submit the real PR, provide a bit more explanation/justification in the comments.

**Pull request merged**